x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/AbstractConfigurationFunctionTestCase.java

```diff
@@ -27,10 +26,25 @@ public abstract class AbstractConfigurationFunctionTestCase extends AbstractScal
 
     @Override
     protected Expression build(Source source, List<Expression> args) {
-        return buildWithConfiguration(source, args, EsqlTestUtils.TEST_CFG);
+        return buildWithConfiguration(source, args, testCase.getConfiguration());
     }
 
-    static Configuration randomConfiguration() {
+    public void testSerializationWithConfiguration() {
+        Configuration config = randomConfiguration();
+        Expression expr = buildWithConfiguration(testCase.getSource(), testCase.getDataAsFields(), config);
+
+        assertSerialization(expr, config);
+
+        Configuration differentConfig;
+        do {
+            differentConfig = randomConfiguration();
+        } while (config.equals(differentConfig));
```

Comment:
NIT: This could be simplified with `Configuration differentConfig = randomValueOtherThan(config, () -> randomConfiguration());`